src/app/features/reporting/reporting/reporting.component.html

```diff
@@ -2,91 +2,25 @@
-    i18n="Button if no reports are configured yet"
-  >
-    Ask for a setup call
-  </a>
-</div>
+<app-select-report
+  [reports]="availableReports"
+  [loading]="loading"
+  [exportableData]="exportableData"
+  (calculateClick)="calculateResults($event.report, $event.from, $event.to)"
+></app-select-report>
 
-<div *ngIf="availableReports?.length > 0"
-  class="work-panel">
-  <div fxLayout="row wrap" fxLayoutGap="1rem">
-    <mat-form-field>
-      <mat-label i18n>Select Report</mat-label>
-      <mat-select [(ngModel)]="selectedReport" [disabled]="loading">
-        <mat-option *ngFor="let report of availableReports" [value]="report">
-          {{ report.title }}
-        </mat-option>
-      </mat-select>
-    </mat-form-field>
+<app-report-row *ngIf="data?.length > 0 && mode === 'reporting'" [rows]="data"></app-report-row>
 
-    <mat-form-field>
-      <mat-label i18n>Enter a date range</mat-label>
-      <mat-date-range-input [rangePicker]="picker" [disabled]="loading">
-        <input
-          [(ngModel)]="fromDate"
-          matStartDate
-          i18n-placeholder="Date selection for the reporting"
-          placeholder="Start date"
-        />
-        <input
-          [(ngModel)]="toDate"
-          matEndDate
-          i18n-placeholder="Date selection for the reporting"
-          placeholder="End date"
-        />
-      </mat-date-range-input>
-      <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
-      <mat-date-range-picker #picker></mat-date-range-picker>
-    </mat-form-field>
-  </div>
-
-  <div>
-    <button
-      mat-stroked-button
-      color="accent"
-      class="primary-button"
-      [disabled]="!selectedReport || (!fromDate && !toDate) || loading"
-      (click)="calculateResults()"
-      i18n="Calculate the results for a report"
-      angulartics2On="click"
-      angularticsCategory="Reporting"
-      angularticsAction="calculate"
-      [angularticsLabel]="selectedReport?.title"
-    >
-      Calculate
-    </button>
-    <button
-      mat-stroked-button
-      color="accent"
-      class="primary-button"
-      [appExportData]="exportableTable"
-      [disabled]="!exportableTable"
-      format="csv"
-      filename="report"
-      style="float: right; margin: 4px"
-      angulartics2On="click"
-      angularticsCategory="Reporting"
-      angularticsAction="export"
-      [angularticsLabel]="selectedReport?.title"
-    >
-      <fa-icon
-        icon="download"
-        class="button-icon"
-        aria-label="download csv"
-      ></fa-icon>
-      <ng-container i18n="Button that allows to download a report">
-        Download report
-      </ng-container>
-    </button>
-  </div>
-
-  <mat-progress-bar mode="indeterminate" *ngIf="loading"></mat-progress-bar>
-</div>
-
-<app-report-row *ngIf="results?.length > 0" [rows]="results"></app-report-row>
+<!--using [hidden] here to ensure that MatSort is correctly initialized-->
```

Comment:
"hidden" means the whole component is initialized and updated invisibly though, doesn't it? That would unnecessarily drain performance?

Reply:
how about moving the whole table into an <app-export-data> component - would that allow us to handle sort correctly even when using ngIf on our custom wrapper component?

Reply:
yes, it's a little less performant but updating only happens when data is assigned to the datasource and this only happens when something should actually be displayed. So I don't think performance is an issue here.

Reply:
> how about moving the whole table into an component - would that allow us to handle sort correctly even when using ngIf on our custom wrapper component?

that could work though